app/services/groups/update_service.rb

```diff
@@ -49,7 +49,7 @@ def after_perform(call)
     if new_user_ids.any?
       db_call = ::Groups::AddUsersService
                   .new(call.result, current_user: user)
-                  .call(ids: new_user_ids)
+                  .call(ids: new_user_ids, send_notifications: params.fetch(:send_notifications, true))
```

Comment:
I think this is not necessary. If there is a `:send_notifications` in `params` hash when calling the `Groups::UpdateService`, then its value has already been retrieved in `BaseServices::BaseContracted#perform`: in app/services/base_services/base_contracted.rb:57` there is:

```
    def perform(params = {})
      service_context(send_notifications: (params || {}).fetch(:send_notifications, true)) do
```

So the value of `params[:send_notifications]` is already retrieved from here, and will be used in `app/services/shared/service_context.rb` where `Journal::NotificationConfiguration.with(send_notifications, &)` is called.

Forwarding it to `Groups::AddUsersService` will not do any harm but it will be useless as `Journal::NotificationConfiguration.with` has already been called as part of calling `Groups::UpdateService`.

I tried removing it and the tests still pass.

BTW, I wonder if the code in `BaseServices::BaseContracted` should also remove the `:send_notifications` from the `params` hash. That would mean you do not need to remove it like you did in `BaseServices::Write` with `params.except(:send_notifications)`.

Maybe something like this could work:

```
    def perform(send_notifications: true, **params)
      service_context(send_notifications:) do
      ...
```

No I just tried in pry and there are case where it does not work. One has to do something like:

```
    def perform(params = {})
      send_notifications = params.fetch(:send_notifications, true)
      params = params.except(:send_notifications)
      service_context(send_notifications:) do
      ...
```

or use another method which does the `**` magic

```
    def perform(params = {})
      do_perform(**params)
    end

    def do_perform(send_notifications: true, **params)
      service_context(send_notifications:) do
      ...
```